WIP BXMSDOC-8000 & BXMSDOC-8001: Update Kogito book as per 7.12 STP BAPLs (#72)
hmanwani-rh/kie-docs: doc-content/enterprise-only/getting-started-kogito/proc-kogito-bpmn-model-creating.adoc

| *Metadata Attributes*
| Add any custom metadata attribute name and value that you want to use for custom event listeners, such as a listener to implement some action when a metadata attribute is present.

You can also use this field to configure role-based access to the process. To configure role-based access, set the attribute *Name* to `securityRoles` and set the attribute value to the relevant group or groups defined for the process, such as `employees,managers`.

| *Global Variables*
|  Not supported in {KOGITO}. Global variables are visible to all process instances and assets in a project, and apply only to distributions of the BPMN modeler in {PRODUCT}.
|===
--
. Begin adding components to your new or imported BPMN process model by clicking and dragging one of the BPMN nodes from the left palette:
+
--
.Adding BPMN components
image::KogitoMicroservices/bpmn-drag-nodes.png[]

Although the BPMN modeler displays many BPMN components in the canvas palette, the {PROCESS_ENGINE} in {PRODUCT} currently supports only the following BPMN components:

* *Start events*
** *Start*
** *Start Signal*

> ** *Start Timer*

Reviewer: @cristianonicolai do we already support timers?

Reviewer 2: not in 7.12, should be removed

Author: Removed timers.

** *Start Message*
* *Intermediate events*
** *Intermediate Signal* (catching and boundary)
** *Intermediate Timer* (catching and boundary)
** *Intermediate Message* (catching, boundary, and throwing)
* *End events*
** *End*
** *End Error*
** *End Terminate*
** *End Message*
* *Tasks*
** *Business Rule*
** *User*
** *Service*
** *Script*
* *Subprocesses*
** *Embedded*
** *Reusable*
* *Gateways*
** *Parallel*
** *Event*
** *Exclusive*
** *Inclusive*

--
. In the BPMN modeler canvas, for each new BPMN component that you add, select the new node, and in the upper-right corner of the BPMN modeler, click *Properties* to define the node identity and behavior.
+
--
For more information about BPMN component properties, see {URL_DEVELOPING_PROCESS_SERVICES}#assembly-designing-business-processes[_{DESIGNING_BUSINESS_PROCESSES}_].

For this example, use a business rule task based on a Decision Model and Notation (DMN) decision model as your first activity node.

> This example assumes that you have the following assets in your {PRODUCT} project:

Reviewer 3: This belongs in prereqs:
You have the following assets in your {PRODUCT} project:
...

Author: Here, the assets are described as a specific example.

** *Data Input*: Add a data input with the name `Person`, with the type `org.acme.Person`, and with the source `person`.
** *Data Output*: Add a data output with the name `isAdult`, with the type `Boolean`, and with the source `isAdult`.
. In the left palette, select *Gateways* -> *Exclusive*, drag the gateway to the canvas, and link to it from the rule task.
. In the left palette, select *Activities* -> *User*, drag the user task to the canvas, and link to it from the exclusive gateway.
. Select the user task and define the following properties:

* *General*: Name the user task `Special handling for children`.
* *Implementation/Execution*: Set the task name to `ChildrenHandling`, and add a data input with the name `person`, the type `org.acme.Person`, and the source `person`.
. In the left palette, select *End Events* -> *End*, drag two end events to the canvas, and link to one end event from the user task and to the other end event from the exclusive gateway.
. Select the connector that connects the exclusive gateway to the end event and for the *Implementation/Execution* property, set the *Condition Expression* to `Java` and enter the condition `return isAdult == true;`.

> . Select the connector that connects the exclusive gateway to the user task and for the *Implementation/Execution* property, set the *Condition Expression* to `Java` and enter the condition to `return isAdult == false;`

Reviewer 3: ```suggestion
. Select the connector that connects the exclusive gateway to the user task and for the *Implementation/Execution* property, set the *Condition Expression* to `Java` and enter the condition to `return isAdult == false;`.
```

Author: Done

. Save the BPMN process file.
+
--
The following is the BPMN model for applicant age evaluation in this example:

.Example `persons.bpmn2` BPMN process
image::KogitoMicroservices/kogito-bpmn-example-person.png[Image of `persons.bpmn2` process diagram]

You can continue adding or modifying any remaining components and properties of your BPMN process or create a separate example.

> The following are additional BPMN models that are used with the `persons.bpmn2` process as part of the same example application:

Reviewer 3: Look for all instances of 'The following are' and rewrite. Something like:
The following additional BPMN models are used with the `persons.bpmn2` process as part of the same example application:

Author: Done

.Example `orders.bpmn2` process
image::KogitoMicroservices/bpmn-model-example-orders.png[Image of `orders.bpmn2` example process]

.Example `orderItems.bpmn2` process invoked as a subprocess
image::KogitoMicroservices/bpmn-model-example-order-items.png[Image of `orderItems.bpmn` example process]

> As an illustration of a more complex use case, the following is an example BPMN model from a separate mortgage loan application for determining loan approval:

Reviewer 3: ```suggestion
The following example illustrates a more complex use case. It is an example BPMN model from a separate mortgage loan application for determining loan approval:
```

Author: Done
